x/stake/endblock.go

```diff
@@ -280,20 +281,33 @@ func handleRefundStake(ctx sdk.Context, sideChainPrefix []byte, k keeper.Keeper)
 	var refundEvents sdk.Events
 	count := 0
 	boundDenom := k.BondDenom(sideChainCtx)
+	bscSideChainId := k.ScKeeper.BscSideChainId(ctx)
+	unBoundingDelegationSize := k.GetUnbondingSizeInQueue(sideChainCtx, ctx.BlockHeader().Time.Add(k.UnbondingTime(ctx)))
+	if unBoundingDelegationSize >= maxUnboundPerBreatheBlock {
+		ctx.Logger().Info("unBoundingDelegation exceed the maxUnboundPerBreatheBlock")
+		return refundEvents
```

Comment:
If return directly, is it possible never goes to the following logic?

Reply:
undelegation has been refactored, please help review